src/SkyApm.Diagnostics.MassTransit/Observers/MasstransitConsumerObserver.cs

```diff
@@ -0,0 +1,85 @@
+﻿using MassTransit;
+using SkyApm.Common;
+using SkyApm.Config;
+using SkyApm.Diagnostics.MassTransit.Common;
+using SkyApm.Tracing;
+using SkyApm.Tracing.Segments;
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace SkyApm.Diagnostics.MassTransit.Observers
+{
+    public class MasstransitConsumerObserver : IConsumeObserver
+    {
+        private const string OperateNamePrefix = "Masstransit Consumer/";
+
+        private readonly ITracingContext _tracingContext;
+        private readonly IGetComponentUtil _getComponentID;
+        private TracingConfig _tracingConfig;
+        public MasstransitConsumerObserver(ITracingContext tracingContext,
+            IGetComponentUtil getComponentID,
+            IConfigAccessor configAccessor)
+        {
+            _tracingContext = tracingContext;
+            _getComponentID = getComponentID;
+            _tracingConfig = configAccessor.Get<TracingConfig>();
+        }
+        public Task PreConsume<T>(ConsumeContext<T> consumeContext) where T : class
+        {
+            var activity = Activity.Current ?? default;
+
+            var operationName = OperateNamePrefix + activity.OperationName;
+            var context = _tracingContext.CreateLocalSegmentContext(operationName);
+            context.Span.SpanLayer = SpanLayer.DB;
+            context.Span.Component = _getComponentID.GetConsumeComponentID(consumeContext);
+            context.Span.Peer = $"{consumeContext.DestinationAddress.Host}";
+            context.Span.AddTag(Tags.MQ_TOPIC, activity.OperationName);
+            context.Span.AddTag(Tags.MQ_BROKER, consumeContext.DestinationAddress.Host);
+            context.Span.AddTag(MassTags.ExpirationTime, consumeContext.ExpirationTime?.ToString("yyyy-MM-dd HH:mm:ss-fff"));
```

Comment:
`ExpirationTime` is a high cardinality tag, best to use logging

Reply:
Thanks for your comments, will replace that with logging.